src/snowflake/snowpark/files.py

```diff
@@ -122,6 +124,20 @@ def isatty(self) -> None:
         """
         raise NotImplementedError(_DEFER_IMPLEMENTATION_ERR_MSG)
 
+    @classmethod
+    @private_preview(version="1.22.1")
+    def open_new_result(cls, mode: str = "w") -> SnowflakeFile:
```

Comment:
Do you think we need a * here? Would that give us more flexibility in the future? Maybe we can check with python folks?

Reply:
verified with Afroz, it will show if version >= 1.22.1. and the version is getting from [version.py](https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/version.py)

Reply:
No, I mean a * in the function definition. Like in the others.

Reply:
Named arguments must follow bare *, I think for now since mode is a required arguments we can leave it here. once we have some unrequired arguments, we can update it like `(cls, mode, *, [arguments])`, similar as current `open(cls, ..., *,...)`

Reply:
hrm okay, it is probably okay then. I think we monkey patch the other one after this one, so it should still work. Thanks for checking.